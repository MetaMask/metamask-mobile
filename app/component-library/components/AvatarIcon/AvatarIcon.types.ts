import { ViewStyle } from 'react-native';
import {
  BaseAvatarProps,
  BaseAvatarSize,
} from '../BaseAvatar/BaseAvatar.types';
import { IconProps, IconSize } from '../Icon/Icon.types';

/**
 * AvatarIcon component props.
 */
export interface AvatarIconProps extends BaseAvatarProps {
  /**
   * Icon to use.
   */
  icon: IconProps['name'];
}

/**
 * AvatarIcon component style sheet.
 */
export interface AvatarIconStyleSheet {
  base: ViewStyle;
}

/**
 * Style sheet input parameters.
 */
export type AvatarIconStyleSheetVars = Pick<AvatarIconProps, 'style'>;

/**
 * Mapping of IconSize by BaseAvatarSize.
 */
export type IconSizeByAvatarSize = {
  [key in BaseAvatarSize]: IconSize;
};
