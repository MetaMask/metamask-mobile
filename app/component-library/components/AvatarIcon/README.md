# AvatarIcon

AvatarIcon is a component that renders an icon contained within an avatar. This component is based on the [BaseAvatar](../BaseAvatar/BaseAvatar.tsx) component.

## Props

This component extends [BaseAvatarProps](../BaseAvatar/BaseAvatar.types.ts#L17) from `BaseAvatar` component.

### `size`

Enum to select between size variants.

| <span style="color:gray;font-size:14px">TYPE</span>    | <span style="color:gray;font-size:14px">REQUIRED</span> |
| :----------------------------------------------------- | :------------------------------------------------------ |
| [BaseAvatarSize](../BaseAvatar/BaseAvatar.types.ts#L6) | Yes                                                     |

### `icon`

Icon to use.

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> |
| :-------------------------------------------------- | :------------------------------------------------------ |
| [IconName](../Icon/Icon.types.ts#L53)               | Yes                                                     |
