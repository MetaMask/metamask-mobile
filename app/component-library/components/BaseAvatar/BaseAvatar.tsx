/* eslint-disable react/prop-types */
import React from 'react';
import { View } from 'react-native';
import { useStyles } from '../../hooks/useStyles';
import { BaseAvatarProps } from './BaseAvatar.types';
import styleSheet from './BaseAvatar.styles';

const BaseAvatar: React.FC<BaseAvatarProps> = ({ size, style, children }) => {
  const { styles } = useStyles(styleSheet, {
    size,
    style,
  });

  return <View style={styles.container}>{children}</View>;
};

export default BaseAvatar;
