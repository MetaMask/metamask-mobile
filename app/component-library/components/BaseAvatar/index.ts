export { default } from './BaseAvatar';
export { BaseAvatarSize } from './BaseAvatar.types';
