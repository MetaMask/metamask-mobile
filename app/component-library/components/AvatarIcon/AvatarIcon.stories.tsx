import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { select } from '@storybook/addon-knobs';
import { BaseAvatarSize } from '../BaseAvatar';
import { IconName } from '../Icon';
import AvatarIcon from './AvatarIcon';

storiesOf(' Component Library / AvatarIcon', module)
  .addDecorator((getStory) => getStory())
  .add('Default', () => {
    const groupId = 'Props';
    const sizeSelector = select(
      'size',
      BaseAvatarSize,
      BaseAvatarSize.Md,
      groupId,
    );
    const iconSelector = select('name', IconName, IconName.LockFilled, groupId);

    return <AvatarIcon size={sizeSelector} icon={iconSelector} />;
  });
