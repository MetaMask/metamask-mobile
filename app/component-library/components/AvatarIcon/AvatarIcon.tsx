/* eslint-disable react/prop-types */
import React from 'react';
import { useStyles } from '../../../component-library/hooks';
import BaseAvatar, { BaseAvatarSize } from '../BaseAvatar';
import Icon, { IconSize } from '../Icon';
import stylesheet from './AvatarIcon.styles';
import { AvatarIconProps, IconSizeByAvatarSize } from './AvatarIcon.types';

const iconSizeByAvatarSize: IconSizeByAvatarSize = {
  [BaseAvatarSize.Xs]: IconSize.Xs,
  [BaseAvatarSize.Sm]: IconSize.Sm,
  [BaseAvatarSize.Md]: IconSize.Md,
  [BaseAvatarSize.Lg]: IconSize.Lg,
  [BaseAvatarSize.Xl]: IconSize.Xl,
};

const AvatarIcon = ({ size, icon, style, ...props }: AvatarIconProps) => {
  const { styles, theme } = useStyles(stylesheet, { style });
  const iconSize = iconSizeByAvatarSize[size];

  return (
    <BaseAvatar size={size} style={styles.base} {...props}>
      <Icon name={icon} size={iconSize} color={theme.colors.primary.default} />
    </BaseAvatar>
  );
};

export default AvatarIcon;
