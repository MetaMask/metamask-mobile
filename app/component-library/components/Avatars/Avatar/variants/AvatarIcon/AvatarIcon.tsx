/* eslint-disable react/prop-types */

// Third party dependencies.
import React from 'react';

// External dependencies.
import AvatarBase from '../../foundation/AvatarBase';
import { AvatarSize } from '../../Avatar.types';
import Icon from '../../../../Icons/Icon';
import { useStyles } from '../../../../../hooks';

// Internal dependencies.
import stylesheet from './AvatarIcon.styles';
import { AvatarIconProps } from './AvatarIcon.types';
import { ICON_SIZE_BY_AVATAR_SIZE } from './AvatarIcon.constants';

const AvatarIcon = ({
  size = AvatarSize.Md,
  name,
  style,
  iconColor: iconColorProp,
  backgroundColor,
  ...props
}: AvatarIconProps) => {
  const { styles, theme } = useStyles(stylesheet, { style, backgroundColor });
  const iconSize = ICON_SIZE_BY_AVATAR_SIZE[size];
  const iconColor = iconColorProp || theme.colors.primary.default;

  return (
    <AvatarBase size={size} style={styles.base} {...props}>
      <Icon name={name} size={iconSize} color={iconColor} />
    </AvatarBase>
  );
};

export default AvatarIcon;
