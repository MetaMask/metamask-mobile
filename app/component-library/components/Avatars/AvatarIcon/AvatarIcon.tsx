/* eslint-disable react/prop-types */

// Third party dependencies.
import React from 'react';

// External dependencies.
import Avatar, { AvatarBaseSize } from '../AvatarBase';
import Icon from '../../Icon';
import { useStyles } from '../../../../component-library/hooks';

// Internal dependencies.
import stylesheet from './AvatarIcon.styles';
import { AvatarIconProps } from './AvatarIcon.types';
import { ICON_SIZE_BY_AVATAR_SIZE } from './AvatarIcon.constants';

const AvatarIcon = ({
  size = AvatarBaseSize.Md,
  name,
  style,
  ...props
}: AvatarIconProps) => {
  const { styles, theme } = useStyles(stylesheet, { style });
  const iconSize = ICON_SIZE_BY_AVATAR_SIZE[size];

  return (
    <Avatar size={size} style={styles.base} {...props}>
      <Icon name={name} size={iconSize} color={theme.colors.primary.default} />
    </Avatar>
  );
};

export default AvatarIcon;
