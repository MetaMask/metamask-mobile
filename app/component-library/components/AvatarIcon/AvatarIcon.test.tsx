import React from 'react';
import { shallow } from 'enzyme';
import AvatarIcon from './AvatarIcon';
import { BaseAvatarSize } from '../BaseAvatar';
import { IconName } from '../Icon';

describe('AvatarIcon', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <AvatarIcon size={BaseAvatarSize.Lg} icon={IconName.AddSquareFilled} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
