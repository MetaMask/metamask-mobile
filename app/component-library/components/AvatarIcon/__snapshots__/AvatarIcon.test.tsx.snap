// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarIcon should render correctly 1`] = `
<BaseAvatar
  size="40"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#037DD619",
      "justifyContent": "center",
    }
  }
>
  <Icon
    color="#037DD6"
    name="AddSquareFilled"
    size="24"
  />
</BaseAvatar>
`;
